src/battle_ai_main.c

```diff
@@ -754,6 +758,235 @@ static inline void BattleAI_DoAIProcessing(struct AI_ThinkingStruct *aiThink, u3
     aiThink->movesetIndex = 0;
 }
 
+static s32 AI_CheckMoveEffects(u32 battlerAtk, u32 battlerDef, u32 move, s32 score, struct AiLogicData *aiData, u32 predictedMove, bool32 isDoubleBattle)
```

Comment:
This function, `AI_TryToClearStats`, `AI_ShouldCopyStatChanges`, and `AI_ShouldSetUpHazards` should be moved to `battle_ai_util.c`

Sorry for the late review but since merge is imminent I wanted to take a second look. That's all though